### PR TITLE
Update import goals script

### PR DESCRIPTION
**Pull request by kryswisnaskas**
## Description of change
Updates the goal import script to reflect the schema changes. Topics and Roles are no longer looked at since they belong to individual objectives. The status of existing goals is updated, but only if the status from RTTAPA is more recent.

## How to test
Looks at the code, make sure the tests pass.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1045

### Every PR

- [x] Meets issue criteria
- [x] JIRA ticket status updated
- [x] Code is meaningfully tested
- [n/a] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [n/a] API Documentation updated
- [n/a] Boundary diagram updated
- [n/a] Logical Data Model updated
- [n/a] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

File: src/tools/importPlanGoals.js [hunk 1]
```diff
@@ -135,26 +100,50 @@ export default async function importGoals(fileKey, region) {
...
               // eslint-disable-next-line no-console
-              console.log(`Couldn't find grant: ${fullGrant.number}. Exiting...`);
+              logger.error(`Couldn't find grant: ${fullGrant.number}. Exiting...`);
               throw new Error('error');
             }
             const grantId = dbGrant.id;
-            await Goal.findOrCreate({
-              where: { grantId, name: goal.name, isFromSmartsheetTtaPlan: true },
-              defaults: goal,
+            const dbGoal = await Goal.findOne({
```
Comment (AdamAdHocTeam) [inline src/tools/importPlanGoals.js:107]: What if we have multiple goals with the same grant and text but some are marked as complete? Do we need to take into account the goal status?

File: src/tools/importPlanGoals.js [hunk 2]
```diff
@@ -84,39 +81,7 @@ export default async function importGoals(fileKey, region) {
...
-                      if (!cleanRoleTopics.some((e) => e.roleId === roleId
-                                                      && e.topicId === topicId)) {
-                        cleanRoleTopics.push({ roleId, topicId });
-                      }
-                    }
-                  }
-                }
-              }
-            } else {
+            if (column !== 'topics') { // ignore topics
```
Comment (AdamAdHocTeam) [inline src/tools/importPlanGoals.js:84]: As we discussed there is no way to map Topics to objectives.

File: src/tools/importPlanGoals.js (hunk 1 above)
Comment (AdamAdHocTeam) [reply src/tools/importPlanGoals.js:107]: Also assuming we don't care about this field anymore 'isFromSmartsheetTtaPlan'.

File: src/tools/importPlanGoals.js [hunk 3]
```diff
@@ -135,26 +130,46 @@ export default async function importGoals(fileKey, region) {
...
+                },
+                {
+                  // where: { id: dbgoal.id },
+                  individualHooks: true,
+                },
+              );
+              // determine if status needs to be updated
+              await updateStatus(goal, dbGoal);
+            } else {
+              const newGoal = await Goal.create({ grantId, ...goal, isFromSmartsheetTtaPlan: true });
```
Comment (thewatermethod) [inline src/tools/importPlanGoals.js:166]: Could we set "createdVia" here to imported?

File: src/tools/importPlanGoals.js (hunk 1 above)
Comment (kryswisnaskas) [reply src/tools/importPlanGoals.js]: > What if we have multiple goals with the same grant and text but some are marked as complete? Do we need to take into account the goal status?

Yes. The status will only be updated if it is newer.

Comment (kryswisnaskas) [reply src/tools/importPlanGoals.js]: > Also assuming we don't care about this field anymore 'isFromSmartsheetTtaPlan'.

We still would want this for historical reasons.

**Review by hardwarehuman — APPROVED** [review]

**Pull request merged**